packages/@aws-cdk/cloudformation-include/test/yaml-templates.test.ts

```diff
@@ -0,0 +1,382 @@
+    includeTestTemplate(stack, 'long-form-sample.yaml');
+
+    expect(stack).toMatchTemplate(
+      loadTestFileToJsObject('long-form-sample.yaml'),
+    );
+  });
+
+  test('can ingest a yaml template with all short form functions except !GetAtt and !Join and !Sub', () => {
+    includeTestTemplate(stack, 'short-form-functions.yaml');
+
+    expect(stack).toMatchTemplate({
+      "Mappings": {
+        "RegionMap": {
+          "region-1": {
+            "HVM64": "name1",
+            "HVMG2": "name2",
+          },
+        },
+      },
+      "Conditions": {
+        "AlwaysTrueCond": {
+          "Fn::Not": [
+            {
+              "Fn::Equals": [
+                {
+                  "Ref": "AWS::Region",
+                },
+                "completely-made-up-region",
+              ],
+            },
+          ],
+        },
+        "AndCond": {
+          "Fn::And": [
+            {
+              "Condition": "AlwaysTrueCond",
+            },
+            {
+              "Fn::Or": [
+                {
+                  "Condition": "AlwaysTrueCond",
+                },
+                {
+                  "Condition": "AlwaysTrueCond",
+                },
+              ],
+            },
+          ],
+        },
+      },
+      "Resources": {
+        "Vpc": {
+          "Type": "AWS::EC2::VPC",
+          "Properties": {
+            "CidrBlock": {
+              "Fn::If": [
+                "AlwaysTrueCond",
+                {
+                  "Fn::Cidr": [
+                    "192.168.1.1/24",
+                    2,
+                    5,
+                  ],
+                },
+                {
+                  "Fn::Cidr": [
+                    "10.0.0.0/24",
+                    "6",
+                    "5",
+                  ],
+                },
+              ],
+            },
+          },
+        },
+        "Bucket": {
+          "Type": "AWS::S3::Bucket",
+          "Properties": {
+            "BucketName": {
+              "Fn::If": [
+                "AndCond",
+                {
+                  "Fn::FindInMap": [
+                    "RegionMap",
+                    "region-1",
+                    "HVM64",
+                  ],
+                },
+                "Unreachable",
+              ],
+            },
+          },
+        },
+        "Subnet1": {
+          "Type": "AWS::EC2::Subnet",
+          "Properties": {
+            "VpcId": {
+              "Fn::If": [
+                "AlwaysTrueCond",
+                {
+                  "Fn::Split": [
+                    ",",
+                    {
+                      "Fn::ImportValue": "ImportedVpcId",
+                    },
+                  ],
+                },
+                "Unreachable",
+              ],
+            },
+            "CidrBlock": "10.0.0.0/24",
+            "AvailabilityZone": {
+              "Fn::Select": [
+                "0",
+                {
+                  "Fn::GetAZs": "",
```

Comment:
Pass an argument here, this is just a missing opportunity.

Reply:
That's meant to test that it can accept both the empty string and a non-empty string, which we give it later. But, this functionality is already covered in the long-form test, so including it with the short-form is probably a waste of time.